### PR TITLE
msb experiments

### PR DESCRIPTION
**Pull request by theoden8**
so here's some testing code. run `make` and then any of the executables

### CODE REVIEW

File: m42v2.h [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    0x00000003F8DC0383ULL, 0x0000007F3A814490ULL, 0x00007DC5C9CF62A6ULL,
+    0x007F23D3342897ACULL, 0x003FEE36EEE1565CULL, 0x0003FF3E99FCCCC7ULL,
+    0x000003ECFCFAC5FEULL, 0x00000003F97F7453ULL, 0x0000000003F8DC03ULL,
+    0x000000007EFA8146ULL, 0x0000007ED3E2EF60ULL, 0x00007F47243ADCD6ULL,
+    0x007FB65AFABFB3B5ULL,
+  };
+
+  inline int msb(uint64_t b)
+  {
+    return std::bit_width(b) - 1;
```
Comment (sinandredemption) [inline m42v2.h:123]: Please try replacing with:
`return std::bit_width(b | 1) - 1;`

**Comment by sinandredemption** [comment]
Hello @theoden8 

I wanted to inform you that your suggestion has been implemented [in this commit](https://github.com/sinandredemption/M42/commit/c6a939bec5c4ff85c342feabb1a2b31ba4a1a453). Besides including intrins for msb, I also included intrins for byteswap.

Note that you would have to manually define `USE_INTRIN` for enabling the usage of the intrins. C++20 support is automatically detected, without requiring any special pre-processor definitions.

Best
Fahad

**Comment by theoden8** [comment]
Hey,

This looks great! By the way, are you sure this is correct? the `b | 1` seemed necessary.
Another issue: std::bit_width is defined in `<bit>`.

Best,
Kirill

**Comment by sinandredemption** [comment]
The ideas mentioned by @theoden8 have been implemented in the [latest commit.](https://github.com/sinandredemption/M42/commit/d0c10584de27997093bcaca97055f91754339ab6). Therefore, I'm closing this issue.

Thank you @theoden8

**Pull request closed without merging**